cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2660,60 +2616,97 @@ int getPodIndexFromPvcName(String pvcName)  {
 
         @SuppressWarnings("deprecation")
         private final Future<ReconciliationState> getEntityOperatorDescription() {
-            Future<ReconciliationState> fut = Future.future();
+            EntityOperator entityOperator = EntityOperator.fromCrd(kafkaAssembly, versions);
 
-            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<ReconciliationState>executeBlocking(
-                future -> {
-                    try {
-                        EntityOperator entityOperator = EntityOperator.fromCrd(kafkaAssembly, versions);
+            if (entityOperator != null) {
+                EntityTopicOperator topicOperator = entityOperator.getTopicOperator();
+                EntityUserOperator userOperator = entityOperator.getUserOperator();
 
-                        if (entityOperator != null) {
-                            EntityTopicOperator topicOperator = entityOperator.getTopicOperator();
-                            EntityUserOperator userOperator = entityOperator.getUserOperator();
+                final String[] configAnnotation = {""};
 
-                            ConfigMap topicOperatorLogAndMetricsConfigMap = topicOperator != null ?
-                                    topicOperator.generateMetricsAndLogConfigMap(topicOperator.getLogging() instanceof ExternalLogging ?
-                                            configMapOperations.get(kafkaAssembly.getMetadata().getNamespace(), ((ExternalLogging) topicOperator.getLogging()).getName()) :
-                                            null) : null;
+                if (topicOperator != null && userOperator != null) {
+                    if (topicOperator.getLogging() instanceof ExternalLogging) {
+                        return configMapOperations.getAsync(kafkaAssembly.getMetadata().getNamespace(), ((ExternalLogging) topicOperator.getLogging()).getName())
+                                .compose(cm -> {
+                                    ConfigMap topicOperatorLogAndMetricsConfigMap = topicOperator.generateMetricsAndLogConfigMap(cm);
+                                    this.topicOperatorMetricsAndLogsConfigMap = topicOperator.generateMetricsAndLogConfigMap(topicOperatorLogAndMetricsConfigMap);
 
-                            ConfigMap userOperatorLogAndMetricsConfigMap = userOperator != null ?
-                                    userOperator.generateMetricsAndLogConfigMap(userOperator.getLogging() instanceof ExternalLogging ?
-                                            configMapOperations.get(kafkaAssembly.getMetadata().getNamespace(), ((ExternalLogging) userOperator.getLogging()).getName()) :
-                                            null) : null;
+                                    configAnnotation[0] += topicOperatorLogAndMetricsConfigMap.getData().get("log4j2.properties");
 
-                            String configAnnotation = "";
+                                    if (userOperator.getLogging() instanceof ExternalLogging) {
```

Comment:
I do not think this would work. Because of the ExteralLogging is not used in Topic Operator but used in User Operator, this code will be never executed.

Reply:
the current version is checking if both Topic Operator logging <https://github.com/strimzi/strimzi-kafka-operator/blob/6527d7acbb2699e98d0ae49321cd9a204abab817/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java#L2675>   and User Operator logging <https://github.com/strimzi/strimzi-kafka-operator/blob/6527d7acbb2699e98d0ae49321cd9a204abab817/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java#L2680>

Should we be only checking User Operator logging?

Reply:
You need to check both. But right now you have wrapped them into each other. So when `if (topicOperator.getLogging() instanceof ExternalLogging) {` evalates to false, this `if (userOperator.getLogging() instanceof ExternalLogging) {` will be never executed.